pkg/operator/sync_v310_00.go

```diff
@@ -0,0 +1,157 @@
+	_, signingSecretModified, err := manageSigningSecret(c)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "signing-key", err))
+	}
+
+	forceDeployment := operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration
+	if saModified { // SA modification can cause new tokens
+		forceDeployment = true
+	}
+	if signingSecretModified {
+		forceDeployment = true
+	}
+	if configMapModified {
+		forceDeployment = true
+	}
+
+	// our configmaps and secrets are in order, now it is time to create the DS
+	// TODO check basic preconditions here
+	actualDeployment, _, err := ensureDeployment_v310_00_to_00(c, operatorConfig, previousAvailability, forceDeployment)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "deployment", err))
+	}
+	if actualDeployment != nil {
+		versionAvailability.UpdatedReplicas = actualDeployment.Status.UpdatedReplicas
+		versionAvailability.ReadyReplicas = actualDeployment.Status.ReadyReplicas
+		versionAvailability.Generations = []operatorsv1alpha1.GenerationHistory{
+			{
+				Group: "apps", Resource: "Deployment",
+				Namespace: targetNamespaceName, Name: "service-serving-cert-signer",
+				LastGeneration: actualDeployment.ObjectMeta.Generation,
+			},
+		}
+	}
+
+	v1alpha1helpers.SetErrors(&versionAvailability, errors...)
+
+	return versionAvailability, errors
+}
+
+// TODO manage rotation in addition to initial creation
+func manageSigningSecret(c ServiceCertSignerOperator) (*corev1.Secret, bool, error) {
+	secret:= resourceread.ReadSecretV1OrDie(v310_00_assets.MustAsset("v3.10.0/service-serving-cert-signer-controller/signing-secret.yaml"))
+	existing, err := c.corev1Client.Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{})
+	if !apierrors.IsNotFound(err){
+		return existing, false, err
+	}
+
+	ca, err := crypto.MakeCAConfig(serviceServingCertSignerName(), 10)
+	if err != nil{
+		return existing, false, err
+	}
+
+	certBytes := &bytes.Buffer{}
+	keyBytes := &bytes.Buffer{}
+	if err := ca.WriteCertConfig(certBytes, keyBytes); err != nil{
+		return existing, false, err
+	}
+
+	secret.Data["tls.crt"] = certBytes.Bytes()
+	secret.Data["tls.key"] = keyBytes.Bytes()
+
+	return resourceapply.ApplySecret(c.corev1Client, secret)
+}
+
+func ensureConfigMap_v310_00_to_00(c ServiceCertSignerOperator, options scsv1alpha1.ServiceCertSignerOperatorConfigSpec) (*corev1.ConfigMap, bool, error) {
+	// TODO use an unstructured object to merge configs
+	config, err := readServiceServingCertSignerConfig(v310_00_assets.MustAsset("v3.10.0/service-serving-cert-signer-controller/defaultconfig.yaml"))
+	if err != nil {
+		return nil, false, err
+	}
+	configBytes, err := runtime.Encode(unstructured.UnstructuredJSONScheme, config)
+	if err != nil {
+		return nil, false, err
+	}
+
+	requiredConfigMap := resourceread.ReadConfigMapV1OrDie(v310_00_assets.MustAsset("v3.10.0/service-serving-cert-signer-controller/cm.yaml"))
+	requiredConfigMap.Data["controller-config.yaml"] = string(configBytes)
```

Comment:
make the key const

Reply:
> make the key const

I really don't want other places to mirror it.  I did make it a const just to make my point though....